azkaban-web-server/src/main/resources/azkaban/webapp/servlet/velocity/executingflowpage.vm

```diff
@@ -97,13 +97,25 @@
   <div class="page-breadcrumb">
     <div class="container-full">
       <ol class="breadcrumb">
-        <li><a
-            href="${context}/manager?project=${projectName}"><strong>Project</strong> $projectName
-        </a></li>
-        <li><a
-            href="${context}/manager?project=${projectName}&flow=${flowid}"><strong>Flow</strong>
-          $flowid
-        </a></li>
+        <li><a href="${context}/manager?project=${projectName}"><strong>Project</strong> $projectName</a></li>
+
+        #set($ref = "${context}/manager?project=${projectName}&flow=")
+        #foreach( $flow in ${flowlist} )
+          #set($ref = $ref + $flow)
+
+          #if( $foreach.first )
+            #set($flowlabel = "Flow")
+            #set($linkclass = "")
+          #else
+            #set($flowlabel = "")
+            #set($linkclass = "flowlink")
+          #end
+
+          <li class="${linkclass}"><a href="${ref}"><strong>$flowlabel</strong> $flow</a></li>
+
+          #set($ref = $ref + ":")
```

Comment:
To avoid hardcoding the path delimiter and reduce repeating this logic across templates what do you think about creating a map of (node name for label, fully qualified path of node for uri) in the servlets and passing that to the velocity template instead of a list of node names only (`flowlist`).

Reply:
Rather than repeating the code on the servlet side, and altering the logic, I am passing the constant down to the form.